src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -289,6 +289,27 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
                     CommunicationsUtilities.Trace("Successfully connected to created node {0} which is PID {1}", nodeId, msbuildProcess.Id);
                     return new NodeContext(nodeId, msbuildProcess, nodeStream, factory, terminateNode);
                 }
+
+                if (msbuildProcess.HasExited)
+                {
+                    if (Traits.Instance.DebugNodeCommunication)
```

Comment:
I have no opinion, but do you want to put this protection around other places, eg where the string `Successfully connected to created node {0}...` is constructed unnecessarily?

Reply:
Those are all const format strings now so they don't create new strings, right?

Reply:
Ungh -- yes. Can't read :)